Client/Frontend/Settings/AppSettingsOptions.swift

```diff
@@ -683,6 +683,53 @@ class OpenFiftyTabsDebugOption: HiddenSetting {
     }
 }
 
+class UseNewHistoryApiOption: HiddenSetting {
+    let profile: Profile
+
+    override init(settings: SettingsTableViewController) {
+        profile = settings.profile
+        super.init(settings: settings)
+    }
+
+    override var title: NSAttributedString? { switch profile.historyApiConfiguration {
+    case .old:
+        return NSAttributedString(string: "Migrate and Use Places History Apis")
+    case .new:
+        return NSAttributedString(string: "Use Legacy History Apis")
+    }}
+
+    override func onClick(_ navigationController: UINavigationController?) {
+        let browserProfile = profile as? BrowserProfile
+        switch profile.historyApiConfiguration {
+        case .old:
+            Logger.browserLogger.info("Setting the history configuration to new")
+            // We run the migration, then we override the profile to use the new apis
+            Logger.browserLogger.info("Deleting history to prepare for migration")
+            let deferred = browserProfile?.places.deleteEverythingHistory()
+            deferred?.upon { _ in
+                Logger.browserLogger.info("History deleted, about to migrate")
+                browserProfile?.migrateHistoryToPlaces(
+                    migrationConfig: .real,
+                    callback: { result in
+                        Logger.browserLogger.info("Successful Migration took \(result.totalDuration / 1000) seconds")
+
+                    }, errCallback: { _ in
+                        // noop
+                    }
+                )
+                browserProfile?.historyApiConfiguration = .new
+                // We set a user default so users with new configuration never go back
+                UserDefaults.standard.setValue(true, forKey: PrefsKeys.NewPlacesAPIDefaultKey)
+            }
+        case .new:
+            Logger.browserLogger.info("Setting the history configuration to old")
+            browserProfile?.historyApiConfiguration = .old
+            // We set a user default so users with new configuration never go back
```

Comment:
nit; Since in that case we're not going to the new config
```suggestion
            // We set a user default so users will stay in that same configuration
```